src/QuadForm/Torsion.jl

```diff
@@ -878,6 +883,35 @@ function _isometry_degenerate(T::TorQuadMod, U::TorQuadMod)
   return (true, TtoU)
 end
 
+# This is a fallback function to cover the case where T and U are not semiregular
+# and they both do not split their radical quadratic.
+function _isometry_non_split_degenerate(T::TorQuadMod, U::TorQuadMod)
+  Ts, TstoT = snf(T)
+  n = ngens(Ts)
+  u_cand = [[u for u in U if quadratic_product(u) == quadratic_product(t) && order(u) == order(t)] for t in gens(Ts)]
+  waiting = Vector{TorQuadModElem}[[]]
+  while !isempty(waiting)
+    f = pop!(waiting)
+    i = length(f)
+    if i == n
+      return (true, compose(inv(TstoT), hom(Ts, U, f)))
+    end
+
+    t = Ts[i+1]
+    card = prod([order(Ts[k]) for k in 1:(i+1)], init = ZZ(1))
+    for u in u_cand[i+1]
+      if all(k -> u*f[k] == t*Ts[k], 1:i)
+        fnew = [j for j in f]
```

Comment:
looks like a `deepcopy` might be more appropriate?

Reply:
`copy` should be also enough.

Reply:
I had a problem of parents at some point, I thought it was coming from this.. apparently not, I will make the change